src/_C017.ino

```diff
@@ -84,44 +90,58 @@ bool do_process_c017_delay_queue(int controller_number, const C017_queue_element
 
   LoadTaskSettings(element.TaskIndex);
 
-  {
+  byte valueCount = getValueCountFromSensorType(element.sensorType);
+  // char itemNames[valueCount][2];
 
-    byte valueCount = getValueCountFromSensorType(element.sensorType);
-    // char itemNames[valueCount][2];
-    for (byte x = 0; x < valueCount; x++)
-    {
+  const size_t capacity = JSON_ARRAY_SIZE(4) + JSON_OBJECT_SIZE(2) + 4 * JSON_OBJECT_SIZE(3); //Size for esp8266: 513
+  DynamicJsonBuffer jsonBuffer(capacity);
 
+  JsonObject &root = jsonBuffer.createObject();
+  root["request"] = "sender data";
 
-const size_t capacity = JSON_ARRAY_SIZE(1) + JSON_OBJECT_SIZE(2) + JSON_OBJECT_SIZE(3);
-DynamicJsonBuffer jsonBuffer(capacity);
+  JsonArray &data = root.createNestedArray("data");
 
-JsonObject& root = jsonBuffer.createObject();
-root["request"] = "sender data";
+  //In my understanding ESPEasy wont return more than 4 Device Values per task
+  JsonObject &data_0 = data.createNestedObject();
+  data_0["host"] = Settings.Name;                            // zabbix hostname, Unit Name for the ESP easy
+  data_0["key"] = ExtraTaskSettings.TaskDeviceValueNames[0]; //valueName // zabbix item
+  data_0["value"] = atof(element.txt[0].c_str());            // ESPeasy supports only floats
 
-JsonArray& data = root.createNestedArray("data");
+  if (valueCount >= 1)
```

Comment:
Should be >, not >=

Can't you use a for loop for this? You don't keep the objects outside the scope of the if-checks, so I don't see why you need to have these extra if's
The max. number of values is defined as MAX_VARS (have to check in the globals file)

Reply:
Using loops to populate JSON is not straightforward. If I find time I will give it a try.
If the task has only 1 Value (valueCount will be 0) then the if statments wont run and the JSON wont populate any further. Anyway I am testing this 2 hours and it works (I am using Uptime, System Load, Free Heap and Free Stack as task values and also Temp and Humidity. Values are send every 60 sec.

Reply:
Valuecount should not be 0 for a plugin that does deliver data.
If it does, then we have a bug.

Reply:
All the other controllers have lines similar to this
```
        for (byte x = 0; x < valueCount; x++)
        {element.txt[x] = formatUserVarNoCheck(event, x);}
```
This tells me that x starts at 0 

I do the same essentialy check but in a opposite way (I check for >= instead of <)

Reply:
Nope, x does start at 0, which is used as index in arrays.
So if `valueCount == 4`, `x` will loop from 0 ... 3
If valueCount = 0, then no occurrence of x will match, since `0 < 0` yields false.

This means any plugin that does output data (e.g. a display does not output values) should have at least 1 as valueCount.
N.B. 0 is also possible. (for the display plugin for example)

So your check will add `data_0` and `data_1` for a plugin that has 1 output value.

Reply:
Ok I will fix it

Reply:
So after digging arrount getValueCountFromSensorType() returns 4 even if only 1 taskvalue is used.
Example use "Generic - System Info " and use only one value for output configuration and leve the rest blank. The getValueCountFromSensorType() will return 4 instead of 1. I dont know if this is meant to be like that.

Reply:
Depends on whether you did set the number of values and saved it (not just adding the plugin, alter it and save)

Reply:
Would it make sence to ignore values whos value name =""
For this condroller case an empty key does not seem to matter to Zabbix.

EG:
`{"request":"sender data","data":[{"host":"zabbix1","key":"uptime","value":13},{"host":"zabbix1","key":"freeheap","value":20160},{"host":"zabbix1","key":"rssi","value":-72},{"host":"zabbix1","key":"","value":0}]}`
The last part has an empty key

Reply:
Thanks TD-er for the suggestions, I figured out the JSON for loop, after messing around with vectors and arrays... I will leave it overnight reporting every 30sec to the zabbix server, see if it still works in the morning

Reply:
About ignoring values without names... There is already a feature request (can't find it now)  about adding some filter to controllers to select when to ignore values.
One thing could be when names are empty (or use a generated one like `name#1` ) or when a value is NaN.
Maybe for some controllers there can be more reasons to ignore (negative values perhaps?)

So I think we should first add a checkmark option to the controller settings for all controllers and then add some options when to consider "illegal" or "undesired" values. (or maybe let the controller decide which one is applicable)